drivers/watchdog/Kconfig

```diff
@@ -1,23 +1,41 @@
 # Kconfig - Watchdog configuration options
 #
-#
 # Copyright (c) 2015 Intel Corporation
+# Copyright (c) 2017 Nordic Semiconductor ASA
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 
-#
-# Watchdog options
-#
+config WDT_LEGACY_API
+	bool "Use legacy Watchdog API (default)"
+	default y
+	help
+	  Driver and user API is the legacy watchdog API (watchdog_legacy.h).
+	  If unselected, the new API will be used.
+
 menuconfig WATCHDOG
 	bool
 	prompt "Watchdog Support"
-	default n
 	help
 	  Include support for watchdogs.
 
 if WATCHDOG
 
+config SYS_LOG_WDT_LEVEL
+	int
+	prompt "Watchdog Driver Log level"
+	depends on SYS_LOG
+	default 0
+	range 0 4
+	help
+	  Sets log level for Watchdog drivers.
+	  Levels are:
+	  0 OFF, do not write
+	  1 ERROR, only write SYS_LOG_ERR
+	  2 WARNING, write SYS_LOG_WRN in addition to previous level
+	  3 INFO, write SYS_LOG_INF in addition to previous levels
+	  4 DEBUG, write SYS_LOG_DBG in addition to previous levels
+
```

Comment:
I propose to add WDT_DISABLE_AT_BOOT option to Kconfig to allow disabling watchdog at boot. Useful at the initial stage of the project development as well as for many sample applications where we don't explicitly handle watchdogs.

Reply:
@mnkp Should it depend on WATCHDOG?

Reply:
@mnkp You think it should be `default y` or `default n`?

Reply:
I think it should be `default n`. The board defconfig file can still choose to set it to `default y` if deemed required.

Later we should only ensure that all the applications in samples/ directory that don't explicitly handle watchdog have `WDT_DISABLE_AT_BOOT` option enabled.